lib/matplotlib/axes/_base.py

```diff
@@ -1282,7 +1282,7 @@ def set_aspect(self, aspect, adjustable=None, anchor=None):
           etc.
           =====   =====================
         """
-        if aspect in ('equal', 'auto'):
+        if isinstance(aspect, str) and aspect in ('equal', 'auto'):
```

Comment:
It might be better to use `cbook.is_string_like` instead of `is_instance`.  Or at least `six.string_types` in place of `str`.